asset-transfer-basic/application-gateway-java/src/main/java/application/java/App.java

```diff
@@ -0,0 +1,251 @@
+        private static final Path certPath    = cryptoPath.resolve(Paths.get("users", "User1@org1.example.com", "msp", "signcerts", "cert.pem"));
+        private static final Path keyPath     = cryptoPath.resolve(Paths.get("users", "User1@org1.example.com", "msp", "keystore"));
+        private static final Path tlsCertPath = cryptoPath.resolve(Paths.get("peers", "peer0.org1.example.com", "tls", "ca.crt"));
+
+        
+	public static void main(String[] args) throws CommitException, InterruptedException, IOException, CertificateException,InvalidKeyException {
+
+		// connect to the network and invoke the smart contract
+                ManagedChannel channel = newGrpcConnection();
+                Gateway.Builder builder = Gateway.newInstance()
+                    .identity(newIdentity())
+                    .signer(newSigner())
+                    .connection(channel)
+                    //default timeouts for different GRPC calls
+                    .evaluateOptions(CallOption.deadlineAfter(5, TimeUnit.SECONDS))
+                    .endorseOptions(CallOption.deadlineAfter(15, TimeUnit.SECONDS))
+                    .submitOptions(CallOption.deadlineAfter(5, TimeUnit.SECONDS))
+                    .commitStatusOptions(CallOption.deadlineAfter(1, TimeUnit.SECONDS));
+               
+        try (Gateway gateway = builder.connect()) {
+        		Network network = gateway.getNetwork(channelName);
+        		Contract contract = network.getContract(chaincodeName);
+        	
+        			//Initialize the Ledger with some data
+        			System.out.println("InitLedger");
+        			initLedger(contract);
+        			System.out.println();
+                    
+                    //getAllAssets
+                    System.out.println("Get All Assets");
+                    getAllAssets(contract);
+                    System.out.println();
+                    //createAsset
+
+                    System.out.println("Create Asset");
+                    createAsset(contract);
+                    System.out.println();
+
+                    //transferAssetAsync
+                    System.out.println("Transfer Asset asynchronously");
+                    transferAssetAsync(contract);
+                    System.out.println();
+
+                    //readAssetByID
+                    System.out.println("Read Asset By ID");
+                    readAssetById(contract);
+                    System.out.println();                    
+
+                    //updateNonExistentAsset
+                    System.out.println("Update a non-existent Asset");
+                    updateNonExistentAsset(contract);
+                    System.out.println();                    
+
+                    
+
+		} catch(Throwable e){
+		  e.printStackTrace();
+		} finally {
+                    channel.shutdownNow().awaitTermination(5, TimeUnit.SECONDS);
+
+                }
+                System.exit(0);
```

Comment:
Don't need the `system.exit()` here... if it JVM doesn't shutdown it means that the `channel.shutdown()` has problems.

Reply:
okay. noted.